File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let hashmap = {};

    for (let i=0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (complement in hashmap) {
            return [i, hashmap[complement]];
        }
        hashmap[nums[i]] = i;
    }
};

File: 1-two-sum/two-sum.cpp
#include <iostream>
#include <set>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;
        int n = nums.size();

        for (int i =0; i < n; i++) {
            int complement = target - nums[i];
            if (hashmap.count(complement)) {
                return {hashmap[complement], i};
            }
            hashmap[nums[i]] = i;
        }
        return {};

    }
};